tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -208,6 +211,9 @@ sealed trait NativeConfig {
   /** Create a new [[NativeConfig]] with updated resource exclude patterns. */
   def withResourceExcludePatterns(value: Seq[String]): NativeConfig
 
+  /** Create a new [[NativeConfig]] with a updated list of service providers excluded from the final binary */
+  def withDisabledServiceProviders(value: Seq[String]): NativeConfig
```

Comment:
Possibly it can be `Map[String, Seq[String]]` defined as `Map[ServiceName, ProviderPattern]` allowing to use regex/glob patterns for exclusions, and possibly to disable whole not interesing services, eg. `Map("java.charset.spi.CharsetProvider" -> Seq(".*"))`
However it would be more verbose

Reply:
Why opt-out instead of opt-in? e.g. for the `ZipFileSystemProvider` seems more like an opt-out sort of thing.

Reply:
Good point. I don't yet have a strong opinion about making it opt-in or opt-out.
My initial idea was that if you have something on classpath you probably would like to use it, and you'd catch the redundant implementations from the logs and possibly eliminate them. But at the same time we kinda go back to issues of `@EnableReflectiveInstantization`
Going opt-in instead of opt-out would probably move most of the usage issues from compile time (missing deps, failed linking) to runtime (exception when there is no suitable file system providers for zip files). 
Both approaches have in my opinion some benefits and some issues. In each case a good communication with the user via logs would be a priority, especially in area of reporting which providers are available, used or discarded.